subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/PomReader.java

```diff
@@ -129,7 +132,8 @@ public InputSource resolveEntity(String publicId, String systemId)
     private final Element projectElement;
     private final Element parentElement;
 
-    public PomReader(final LocallyAvailableExternalResource resource) throws SAXException {
+    public PomReader(final LocallyAvailableExternalResource resource, Map<String, String> childPomProperties) throws SAXException {
+        setPomProperties(childPomProperties);
```

Comment:
The maybeSetGavProperties and setParentPomProperties put in properties
something else than what is inside the <properties> tag of the pom.
The purpose of the set of property keys is to record what is in properties, as these are the only ones that should be overridden in parent/grand-parent poms.